miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/view/EditServiceRecordController.java

```diff
@@ -58,26 +55,45 @@ public class EditServiceRecordController {
   private ObjectMapper mapper;
 
   @GetMapping(value = "/{recordId}")
-  public ModelAndView viewServiceRecord(@PathVariable(value = "recordId") Long recordId, ModelMap model) throws IOException {
+  public ModelAndView viewServiceRecord(@PathVariable(value = "recordId") Long recordId, Long instrumentId,
+      ModelMap model)
+      throws IOException {
     ServiceRecord record = serviceRecordService.get(recordId);
+    Instrument instrument = instrumentService.get(instrumentId);
     if (record == null) {
       throw new NotFoundException("No service found for ID " + recordId.toString());
     }
-    return showPage(record, model);
+    return showPage(record, instrument, model);
   }
-  
-  @GetMapping(value = "/new/{instrumentId}")
-  public ModelAndView newServiceRecord(@PathVariable(value = "instrumentId") Long instrumentId, ModelMap model) throws IOException {
+
+  @GetMapping(value = "/new/{recordId}")
+  public ModelAndView newServiceRecord(@PathVariable(value = "recordId") Long recordId,
+      Long instrumentId, ModelMap model)
+      throws IOException {
     Instrument instrument = instrumentService.get(instrumentId);
     if (instrument == null) {
       throw new NotFoundException("No instrument found for ID " + instrumentId.toString());
     }
     ServiceRecord record = new ServiceRecord();
-    record.setInstrument(instrument);
-    return showPage(record, model);
+    instrumentService.addServiceRecord(record, instrument);
```

Comment:
This should just be setting up the UI to create a service record. It shouldn't be created in the database at this point